paparazzi/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -80,7 +84,11 @@ class PaparazziPlugin : Plugin<Project> {
         project.tasks.named("merge${variantSlug}Assets") as TaskProvider<MergeSourceSetFolders>
       val mergeAssetsOutputDir = mergeAssetsProvider.flatMap { it.outputDir }
       val reportOutputDir = project.layout.buildDirectory.dir("reports/paparazzi")
-      val snapshotOutputDir = project.layout.projectDirectory.dir("src/test/snapshots")
+
+      val snapshotOutputDir: File =
+        extension.snapshotRootDirectory.orNull?.asFile
+          ?: project.layout.projectDirectory.dir("src/test/snapshots").asFile
+      snapshotOutputDir.mkdirs()
```

Comment:
No-one should be accessing the file system during configuration, I don't think this is configuration cache compatible, and also unnecessary I/O. All I/O should happen during Task execution.

Reply:
This means that we don't even have access to `File` objects... we don't need to, see `snapshotDirectory` is a `DirectoryProperty` and the other too. If we create the folder, it should be somewhere near `it.write(snapshotDirectory.get().asFile.invariantSeparatorsPath)`

Reply:
oh, wait, the File stuff applies, but the folder creation is suspect, since this is input... if it doesn't exist that's a failure?

Reply:
There is an open issue about this on Gradle

https://github.com/gradle/gradle/issues/2016

Another option is using the workaround mentioned there, so:

```kotlin
@get:InputFiles
abstract val snapshotDirectory: DirectoryProperty
```